src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -2432,7 +2430,9 @@ Symbol findIdentInternal(Env<AttrContext> env, Name name, KindSelector kind) {
 
         if (kind.contains(KindSelector.TYP)) {
             sym = findType(env, name);
-
+            if ((sym.flags() & ANONYMOUS_MAIN_CLASS) != 0) {
```

Comment:
If `sym` is also `SYNTHTIC` (which it is) - do you need this? In what case? E.g. if there's a legitimate case for this I'm wondering if we should just test for `SYNTHETIC` rather than `ANONYMOUS_MAIN_CLASS`.

Reply:
Yes, this is a remnant from earlier work.